.github/workflows/changelog_pr_fields.sh

```diff
@@ -0,0 +1,98 @@
+         #Extracting the values for each field  
+
+          pr_body=$(jq -r ".pull_request.body" "$GITHUB_EVENT_PATH")| 
+
+          # mandatory changelog_start and changelog_end
+          
+          changelog_start="<!-- [DO NOT REMOVE-USED BY GH ACTION] CHANGELOG START -->"
+          changelog_end="<!-- [DO NOT REMOVE-USED BY GH ACTION] CHANGELOG END -->"
+
+          if [[ "$pr_body" != *"$changelog_start"* ]]; then
+            echo "[DO NOT REMOVE-USED BY GH ACTION] CHANGELOG START is missing: $changelog_start"
+            exit 1
+          fi
+
+          if [[ "$pr_body" != *"$changelog_end"* ]]; then
+            echo "[DO NOT REMOVE-USED BY GH ACTION] CHANGELOG END is missing: $changelog_end"
+            exit 1
+          fi
+
+          # Extract and trim each field using pattern matching
+
+          description=$(echo "$pr_body" | awk '/## Changelog/{flag=1; next} /##/{flag=0} flag' | grep -oP '(?<=- \*\*Description:\*\* ).*' | sed 's/^[[:space:]]*//;s/[[:space:]]*$//') echo "$description"
+          product_impact=$(echo "$pr_body" | awk '/## Changelog/{flag=1; next} /##/{flag=0} flag' | grep -oP '(?<=- \*\*Product Impact:\*\* ).*' | sed 's/^[[:space:]]*//;s/[[:space:]]*$//') echo "$product_impact"
+          addresses=$(echo "$pr_body" | awk '/## Changelog/{flag=1; next} /##/{flag=0} flag' | grep -oP '(?<=- \*\*Addresses:\*\* ).*' | sed 's/^[[:space:]]*//;s/[[:space:]]*$//') echo "$addresses"
+          components=$(echo "$pr_body" | awk '/## Changelog/{flag=1; next} /##/{flag=0} flag' | grep -oP '(?<=- \*\*Components:\*\* ).*' | sed 's/^[[:space:]]*//;s/[[:space:]]*$//') echo "$components"
+          breaking=$(echo "$pr_body" | awk '/## Changelog/{flag=1; next} /##/{flag=0} flag' | grep -oP '(?<=- \*\*Breaking:\*\* ).*' | sed 's/^[[:space:]]*//;s/[[:space:]]*$//') echo "$breaking"
+          impact_a11y=$(echo "$pr_body" | awk '/## Changelog/{flag=1; next} /##/{flag=0} flag' | grep -oP '(?<=- \*\*Impact_a11y:\*\* ).*' | sed 's/^[[:space:]]*//;s/[[:space:]]*$//') echo "$impact_a11y"
+          guidance=$(echo "$pr_body" | awk '/## Changelog/{flag=1; next} /##/{flag=0} flag' | grep -oP '(?<=- \*\*Guidance:\*\* ).*' | sed 's/^[[:space:]]*//;s/[[:space:]]*$//') echo "$guidance"
+
+         # Default values for each field(According to PR template)
+
+          default_description="Summary of change(s)"
```

Comment:
I wonder if we could read these values from the Github PR template instead? We could make the Python script used to generate the changelog reusable, and then use it here to parse the Github PR template, and extract the default values from there.

Reply:
ok..as you said the python file is more maintainable ,it would make the code more dynamic .But i will need some days to work on it .

Reply:
No problem @RONAK-AI647, take your time. We appreciate it.